scripts/common.py

```diff
@@ -81,3 +82,23 @@ def inner():
             return False
 
     return call_rate_limit_aware(inner)
+
+
+def get_wrappers(smkfile):
+    wrappers = {}
+    for match in re.finditer(r"\n\s*wrapper:\s*(.+)\n", smkfile):
```

Comment:
CodeRabbitAI might have a point here, regarding wrappers at the end of the file without a newline afterwards. But I think the following should work, right?

```suggestion
    for match in re.finditer(r"^*wrapper:\s*(.+)$", smkfile):
```
[According to the docs, this should match the start and end of a line in multiline mode.](https://docs.python.org/3/library/re.html#regular-expression-syntax)

Reply:
I also thought about the comment from code rabbit, but since wrappers are only parsed for standardized workflows, it makes sense in my opinion to be strict about the formatting. In that sense, wrapper statements are expected to appear on a separate line, and have a trailing new line after it. The regex scans the entire smk file as a string, not line-wise, so using `^` and `$` will not work as it matches start and end of the file, not the line. My take here is to adhere to the standards we try to have with `snakefmt`, and just disregard strangely formatted cases.

Reply:
OK, that sounds like a valid point. And I also double-checked that `snakefmt` does add a `\n` at the end of the last line. And that `\s` also matches `\n` in Python regular expressions. So this should indeed be fine!